storage/ndb/src/common/mgmcommon/thr_config.cpp

```diff
@@ -887,32 +587,80 @@ THRConfig::do_parse(unsigned realtime,
   {
     add(T_RECV, realtime, spintime);
   }
+  Uint32 num_query_instances =
+    ldm_threads +
+    tc_threads +
+    recv_threads +
+    main_threads +
+    rep_threads;
+  Uint32 calc_num_cpus = num_query_instances + send_threads;
+  require(calc_num_cpus <= num_cpus);
+
   struct ndb_hwinfo *hwinfo = Ndb_GetHWInfo(false);
   if (hwinfo->is_cpuinfo_available && num_cpus == 0)
```

Comment:
You reset num_cpus from zero(if it was) to a non-zero value in compute_automatic_thread_config() (Line 401), so it appears that this if condition would never be satisfied.

Reply:
Great, I noted this bug when I was benchmarking, it means I didn't get any CPU locking, great catch.